api/src/main/java/com/example/controllers/DrillController.java

```diff
@@ -0,0 +1,52 @@
+package com.example.controllers;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Created by viki on 05.05.16.
+ */
+
+@RestController
+public class DrillController {
+
+    /**
+     *  Function, that gives list of all drills in database.
+     *
+     *  @return drills - ArrayList of type Drill
+     */
+    @RequestMapping(value="/drill", method= RequestMethod.GET)
+    public Collection<Object> getAllDrills() {
+        List<Object> drills = new ArrayList<Object>();
+        // get all drills from DB by calling model layer
+        return drills;
+    }
+
+    /**
+     * Function, that returns drill in databse with given ID. If the ID is not given or do not corrspond to any drill in database, function returns null.
+     *
+     * @param drillId - ID of relevant drill
+     * @return drill - object of type Drill with given id, null if drill with the id does not exist
+     */
+    @RequestMapping(value="/drill", method= RequestMethod.POST)
+    public Object getDrillById(@RequestParam(value="drill_id") Long drillId ) {
```

Comment:
To `drillId` (a celkově všechny idčka v aplikaci) bych ideálně chtěl mít jako [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier), java na to má určitě nějakou knihovnu :)